app/controllers/providers/client_received_legal_helps_controller.rb

```diff
@@ -0,0 +1,36 @@
+module Providers
+  class ClientReceivedLegalHelpsController < BaseController
+    include ApplicationDependable
+    include Steppable
+    include SaveAsDraftable
+    before_action :authorize_legal_aid_application
+
+    def show
+      @form = MeritsAssessments::ClientReceivedLegalHelpForm.new(model: merits_assessment)
+    end
+
+    def update
+      @form = MeritsAssessments::ClientReceivedLegalHelpForm.new(client_received_legal_help_params.merge(model: merits_assessment))
+
+      if @form.save
```

Comment:
A suggestion. If you add this private method:
```ruby
def form
  @form ||= MeritsAssessments::ClientReceivedLegalHelpForm.new(model: merits_assessment)
end
```
I think the code of both the `show` and `update` methods can be simplified a lot:

```ruby
def show
  form
end

def update
  if form.update(client_received_legal_help_params)
    continue_or_save_draft
  else
    render :show
  end
end
```

Reply:
But the code of `show` and `update` is not exactly the same.
the form in `update` also needs the `params`

```ruby
# show
MeritsAssessments::ClientReceivedLegalHelpForm.new(model: merits_assessment)

# update
MeritsAssessments::ClientReceivedLegalHelpForm.new(client_received_legal_help_params.merge(model: merits_assessment))
```